# < F-Rated Korean Films List >

항목 분류 참고: http://f-rated.org/f-rated-films/

In [1]:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

%matplotlib inline
plt.rcParams['font.family'] = 'NanumGothic'
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
pd.set_option('float_format', '{:f}'.format)
warnings.filterwarnings('ignore')

In [25]:
df = pd.read_csv('../01.Data_Collection/films_updated.csv')
df.shape

(690, 38)

In [17]:
# df.columns

In [18]:
# f_films = df[df['F-rated']!=0][['영화명', '영화코드', '개봉일', '장르', 'F-rated', 
#                                 '감독', '대표감독', '감독코드', '감독성별', 
#                                 '작가', '대표작가', '작가코드', '작가성별', 
#                                 '배우', '주연배우', '배우코드', '배우성별']]

In [19]:
# len(f_films)

In [20]:
# 반절 이상이 F-0
# 686-337

In [14]:
# F-0만 제외한 리스트를 엑셀파일로 저장
# f_films.to_excel('../01.Data_Collection/poster.xlsx', index=False)

엑셀파일 작업 내용
1. 특정 장르 삭제 : 총 35개
    * 가족, 뮤지컬 2
    * 공연 2
    * 기타 1
    * 뮤지컬 1
    * 애니메이션 30
2. 장르가 여러 개 써있는 영화가 많아서 1영화당 1장르로 맞춤 -> 근데 장르 종류가 너무 많고 장르당 영화개수의 차이가 커서 몇 개씩 합칠 필요 있음.
    * 드라마 +23
    * 멜로/로맨스 +12
    * 다큐멘터리 +4
    * 공포(호러) +4
    * 스릴러 +8
    * 코미디 +15
    * 범죄 + 11
    * 미스터리 +12
    * 사극 +4
    * 액션 +2
    * 판타지 +1
3. 

영화별 출연배우 성별 확인 작업

In [23]:
actor = pd.read_csv('../01.Data_Collection/actor_detail.csv')
actor.head()

movieCd      movieNm moviePartNm  peopleCd peopleNm sex
0  20197922  다만 악에서 구하소서          배우  10090290      황정민  남자
1  20192194       인질(가제)          배우  10090290      황정민  남자
2  20179101           공작          배우  10090290      황정민  남자
3  20161481          군함도          배우  10090290      황정민  남자
4  20177063      군함도 감독판          배우  10090290      황정민  남자

In [24]:
actor[actor['movieCd']==20181103]

movieCd movieNm moviePartNm  peopleCd peopleNm sex
132215  20181103      벌새          배우  20320946      박지후  여자
132222  20181103      벌새          배우  20126450      김새벽  여자
132251  20181103      벌새          배우  10062134      정인기  남자
132291  20181103      벌새          배우  10056247      이승연  여자
132328  20181103      벌새          배우  20320858      박수연  여자
132355  20181103      벌새          배우  20280689      손용범  남자
132364  20181103      벌새          배우  20328356      박서윤  여자
132369  20181103      벌새          배우  20113258      길해연  여자
132403  20181103      벌새          배우  20306130      손상연  남자
132405  20181103      벌새          배우  20213907      설혜인  여자
136764  20181103      벌새          배우  10062134      정인기  남자
137519  20181103      벌새          배우  20126450      김새벽  여자
138657  20181103      벌새          배우  20113258      길해연  여자

# Triple F-Rated Films

In [16]:
triple = film[film['F-rated']==3]

In [17]:
# 최신개봉작부터 정렬
triple = triple.sort_values('개봉일', ascending=False)

In [26]:
len(triple)

49

In [27]:
triple

영화명      영화코드         개봉일              장르  F-rated  \
685              벌새  20181103  2019-08-29             드라마        3   
682             우리집  20190703  2019-08-22          드라마,가족        3   
681       나만 없어 고양이  20191105  2019-08-22      가족,판타지,드라마        3   
677       밤의 문이 열린다  20186645  2019-08-15         판타지,드라마        3   
626      막다른 골목의 추억  20189803  2019-04-04          멜로/로맨스        3   
618          선희와 슬기  20187301  2019-03-27             드라마        3   
612        1919 유관순  20196881  2019-03-14           다큐멘터리        3   
611           히치하이크  20174042  2019-03-14              가족        3   
581          버블 패밀리  20174244  2018-12-20        다큐멘터리,가족        3   
578          어른이 되면  20184642  2018-12-13           다큐멘터리        3   
564              영주  20177341  2018-11-22             드라마        3   
553             밤치기  20177981  2018-11-01          멜로/로맨스        3   
551      폴란드로 간 아이들  20189281  2018-10-31       다큐멘터리,드라마        3   
543             미쓰백  20189067  2018-10-11             드라마        3   
529             나비잠  20175587  2018-09-06          멜로/로맨스        3   
504         안나 카레니나  20185484  2018-06-29             뮤지컬        3   
503         너와 극장에서  20175865  2018-06-28             드라마        3   
479             수성못  20174265  2018-04-19         드라마,코미디        3   
465             소공녀  20173962  2018-03-22      멜로/로맨스,드라마        3   
443          누에치던 방  20160484  2018-01-31             드라마        3   
435          피의 연대기  20175930  2018-01-18           다큐멘터리        3   
434      파란입이 달린 얼굴  20158358  2018-01-18             드라마        3   
428       나의 연기 워크샵  20160477  2017-12-28             드라마        3   
392            유리정원  20164481  2017-10-25        미스터리,드라마        3   
380        여배우는 오늘도  20172642  2017-09-14         드라마,코미디        3   
358          불온한 당신  20154427  2017-07-20           다큐멘터리        3   
336  고려 아리랑: 천산의 디바  20164041  2017-05-25           다큐멘터리        3   
306              눈길  20158799  2017-03-01             드라마        3   
291              문영  20158526  2017-01-12             드라마        3   
280          비치온더비치  20161806  2016-12-08          멜로/로맨스        3   
276      미씽: 사라진 여자  20153741  2016-11-30            미스터리        3   
268             연애담  20159286  2016-11-17          멜로/로맨스        3   
246        할머니의 먼 집  20158359  2016-09-29           다큐멘터리        3   
245              물숨  20162969  2016-09-29           다큐멘터리        3   
237            달빛궁궐  20166541  2016-09-07  애니메이션,판타지,어드벤처        3   
215          홀리워킹데이  20150258  2016-06-30           다큐멘터리        3   
211       우리 연애의 이력  20154803  2016-06-29          멜로/로맨스        3   
209          비밀은 없다  20140704  2016-06-23             스릴러        3   
207             우리들  20153450  2016-06-16             드라마        3   
206          소녀와 여자  20155485  2016-06-16           다큐멘터리        3   
189              초인  20155388  2016-05-05          멜로/로맨스        3   
166       설행_눈길을 걷다  20158822  2016-03-03             드라마        3   
162            소꿉놀이  20158161  2016-02-25           다큐멘터리        3   
157            좋아해줘  20150977  2016-02-17      멜로/로맨스,코미디        3   
109             거짓말  20141608  2015-10-29             드라마        3   
79             오늘영화  20143717  2015-08-20       멜로/로맨스,기타        3   
61              파스카  20132801  2015-07-09             드라마        3   
58              마돈나  20159161  2015-07-02             드라마        3   
40        잡식가족의 딜레마  20148676  2015-05-07           다큐멘터리        3   

                  감독 대표감독      감독코드 감독성별  \
685              김보라  김보라  20110847   여자   
682              윤가은  윤가은  20113672   여자   
681          복운석,신혜진  신혜진  10040701   여자   
677              유은정  유은정  20152055   여자   
626              최현영  최현영  20111804   여자   
618              박영주  박영주  20183969   여자   
612              신상민  신상민  20318581   여자   
611              정희재  정희재  20186944   여자   
581              마민지  마민지  20276651   여자   
578              장혜영  장혜영  20304969   여자   
564              차성덕  차성덕  1007

# F-Rated Films (F-rated 컬럼값이 2 또는 1)

## Directed by Women
Triple 제외

In [20]:
directed = film[(film['감독성별']=='여자') & (film['F-rated']!=3)].sort_values('개봉일', ascending=False)

In [21]:
directed

영화명      영화코드         개봉일                 장르  F-rated  \
678           이타미 준의 바다  20190470  2019-08-15              다큐멘터리        2   
657             한낮의 피크닉  20183727  2019-07-04                드라마        2   
647              보희와 녹양  20181198  2019-05-29                드라마        2   
646            우리 지금 만나  20191302  2019-05-29                드라마        2   
635  굿바이 마이 러브NK: 붉은 청춘  20174284  2019-05-02              다큐멘터리        2   
621                  생일  20183096  2019-04-03                드라마        2   
617             썬키스 패밀리  20166301  2019-03-27      가족,멜로/로맨스,코미디        2   
615                   돈  20176103  2019-03-20                 범죄        2   
606                  포항  20148617  2019-02-27                드라마        2   
586                 말모이  20184105  2019-01-09                드라마        2   
573             번개맨의 비밀  20183650  2018-12-06             가족,뮤지컬        2   
549              집의 시간들  20172388  2018-10-25              다큐멘터리        2   
538   극장판 뽀잉: 슈퍼 변신의 비밀  20189746  2018-09-20              애니메이션        1   
523                어른도감  20181821  2018-08-23            코미디,드라마        2   
496             탐정: 리턴즈  20178161  2018-06-13             코미디,범죄        1   
474                 머니백  20165045  2018-04-12             액션,드라마        2   
469                  덕구  20181561  2018-04-05                드라마        2   
458            바나나쏭의 기적  20170422  2018-03-08              다큐멘터리        2   
452             리틀 포레스트  20170841  2018-02-28                드라마        2   
420                산상수훈  20170272  2017-12-07                드라마        2   
400                 부라더  20172603  2017-11-02                코미디        2   
401                 메소드  20173612  2017-11-02                드라마        2   
385          다시 태어나도 우리  20169463  2017-09-27              다큐멘터리        2   
379              시인의 사랑  20178941  2017-09-14                드라마        2   
375              안녕 히어로  20169270  2017-09-07              다큐멘터리        2   
357              파밍 보이즈  20160746  2017-07-13              다큐멘터리        2   
325       서서평, 천천히 평온하게  20173301  2017-04-26              다큐멘터리        2   
307                  해빙  20150971  2017-03-01                스릴러        2   
302               싱글라이더  20161723  2017-02-22                드라마        2   
282        당신, 거기 있어줄래요  20152371  2016-12-14            판타지,드라마        2   
269           야근 대신 뜨개질  20154428  2016-11-17              다큐멘터리        2   
262                  시소  20142706  2016-11-10              다큐멘터리        2   
208                  삼례  20158821  2016-06-23            드라마,판타지        2   
172                  커터  20152746  2016-03-30             범죄,드라마        2   
168                  히야  20161722  2016-03-10                드라마        2   
160                  순정  20151021  2016-02-24                드라마        1   
148                  하프  20151224  2016-01-21                드라마        2   
142           나를 잊지 말아요  20142641  2016-01-07             멜로/로맨스        2   
127    앨리스: 원더랜드에서 온 소년  20141002  2015-12-10  판타지,공포(호러),멜로/로맨스        1   
126               나쁜 나라  20156047  2015-12-03              다큐멘터리        2   
104           특종: 량첸살인기  20145504  2015-10-22                스릴러        2   
99                   비밀  20156553  2015-10-15           미스터리,드라마        1   
50             마이 페어 웨딩  20141705  2015-06-04              다큐멘터리        2   
34           반짝이는 박수 소리  20148982  2015-04-23              다큐멘터리        2   
8               생각보다 맑은  20145202  2015-01-22              애니메이션        2   

              감독  대표감독      감독코드 감독성별  \
678          정다운   정다운  20324792   여자   
657  강동완,김한라,임오정   김한라  20177885   여자   
647          안주영   안주영  20191668   여자   
646  김서윤,강이관,부지영   김서윤  20318832   여자   
635          김소영   김소영  10004709   여자   
621          이종언   이종언  20159971   여자   
617          김지혜   김지혜  20239886   여자   
615          박누리   박누리  20151353   여자   
606          모현신   모현신  10026500   여자   
586          엄유나   엄유나  20187477   여자   
573          강유선   강유선  20

수정할 것
1. 657번 '한낮의 피크닉' 대표감독, 대표작가
2. 269번 '야근 대신 뜨개질' 주연배우

## Written by Women
Triple, Directed by Women 제외

In [22]:
written = film[(film['작가성별']=='여자') & (film['F-rated']!=3) & (film['감독성별']!='여자')].sort_values('개봉일', ascending=False)

In [23]:
written

영화명      영화코드         개봉일             장르  \
683                     유열의 음악앨범  20181416  2019-08-28     멜로/로맨스,드라마   
680                           변신  20184543  2019-08-21     공포(호러),스릴러   
662       극장판 샤이닝스타:새로운 루나퀸의 탄생!  20193013  2019-07-18          애니메이션   
644                       시민 노무현  20191141  2019-05-23          다큐멘터리   
642                       어린 의뢰인  20199448  2019-05-22         드라마,가족   
639                        물의 기억  20190725  2019-05-15      다큐멘터리,드라마   
631                      뷰티플 마인드  20188083  2019-04-18          다큐멘터리   
620                   아픈 만큼 사랑한다  20198043  2019-04-03          다큐멘터리   
598                        시인 할매  20189649  2019-02-05          다큐멘터리   
589                    그대 이름은 장미  20166602  2019-01-16            코미디   
580                          마약왕  20168773  2018-12-19         범죄,드라마   
574                          다영씨  20183390  2018-12-06        드라마,코미디   
569                       소녀의 세계  20165562  2018-11-29         멜로/로맨스   
565                         하나식당  20187303  2018-11-22            드라마   
561               번 더 스테이지: 더 무비  20183050  2018-11-15             기타   
550                       완벽한 타인  20177538  2018-10-31        드라마,코미디   
521                          소성리  20174348  2018-08-16          다큐멘터리   
515            극장판 헬로카봇 : 백악기 시대  20186501  2018-08-01          애니메이션   
502                    나와 봄날의 약속  20189802  2018-06-28       미스터리,판타지   
501                         허스토리  20172469  2018-06-27            드라마   
489                           독전  20179383  2018-05-22          범죄,액션   
488                5.18 힌츠페터 스토리  20183381  2018-05-17          다큐멘터리   
480                      파파 오랑후탄  20175046  2018-04-19          다큐멘터리   
470                   번개맨과 신비의 섬  20182021  2018-04-05         가족,뮤지컬   
467                        7년의 밤  20154908  2018-03-28        스릴러,드라마   
461                       엄마의 공책  20164389  2018-03-15         드라마,가족   
453                           궁합  20150969  2018-02-28        코미디,드라마   
447              흥부: 글로 세상을 바꾼 자  20176121  2018-02-14             사극   
444              조선명탐정: 흡혈괴마의 비밀  20179205  2018-02-08    사극,어드벤처,코미디   
441                     기생-꽃의 고백  20180321  2018-01-25          다큐멘터리   
409                      아이엠 호프맨  20175047  2017-11-16          다큐멘터리   
402                    내게 남은 사랑을  20177342  2017-11-02            드라마   
396                     뷰티풀 투모로우  20175208  2017-10-26            드라마   
395                     내 친구 정일우  20178461  2017-10-26          다큐멘터리   
382                     아이 캔 스피크  20174142  2017-09-21            코미디   
360                        택시운전사  20162869  2017-08-02            드라마   
356                     올 리브 올리브  20169269  2017-07-13          다큐멘터리   
345                   나는 고양이로소이다  20175601  2017-06-08          다큐멘터리   
340                       바람의 춤꾼  20176041  2017-06-06          다큐멘터리   
326             극장판 또봇: 로봇군단의 습격  20174165  2017-04-27          애니메이션   
319                        아빠는 딸  20151227  2017-04-12            코미디   
318                       다시, 벚꽃  20172381  2017-04-06          다큐멘터리   
317                          어느날  20167901  2017-04-05         멜로/로맨스   
284                          마스터  20161725  2016-12-21          액션,범죄   
272                           혼자  20155387  2016-11-24            스릴러   
270                            형  20154661  2016-11-23        코미디,드라마   
265                       가려진 시간  20154905  2016-11-16        드라마,판타지   
263                      오 마이 파파  20162927  2016-11-10          다큐멘터리   
254        폭력의 법칙: 나쁜 피 두 번째 이야기  20162970  2016-10-20            드라마   
236                   고산자, 대동여지도  20159648  2016-09-07             사극   
217            극장판 프리즘스톤 올스타 셀렉션  20165521  2016-07-07  애니메이션,드라마,판타지   
213                       빅뱅 메이드  20165042  2016-06-30          다큐멘터리   
210                       굿바이 싱글  20150972  2016-06-29        코미디,드라마   
196                         계춘할망  20156559  2016-05-19         가족,드라마   
193           

## Starring Significant Female Characters (출연배우 데이터의 첫번째 이름이 여성)
Triple, Directed by Women, Written by Women 제외  
F-rated 홈페이지에도 '여자 배우가 중요한 역할을 하는가'는 논란의 여지가 많은 항목으로 관객끼리 서로 의견을 나눠볼 것을 제안하고 있음 -> 포스터에 작은 글씨로 이런 내용도 적어주면 좋을 듯

In [24]:
starring = film[(film['배우성별']=='여자') & (film['F-rated']==1)].sort_values('개봉일', ascending=False)

In [25]:
starring

영화명      영화코드         개봉일  \
676                        암전  20188404  2019-08-15   
673                        려행  20162140  2019-08-08   
672                       김복동  20190741  2019-08-08   
671           캐리tv 러브콘서트 더 무비  20193642  2019-08-07   
666                      레드슈즈  20198323  2019-07-25   
656                     별의 정원  20192689  2019-07-04   
653                       에움길  20184961  2019-06-20   
649                   소은이의 무릎  20176026  2019-05-30   
645                    0.0MHz  20188144  2019-05-29   
638                      배심원들  20187981  2019-05-15   
636                       걸캅스  20184082  2019-05-09   
633           뽀로로 극장판 보물섬 대모험  20198183  2019-04-25   
630                    크게 될 놈  20198982  2019-04-18   
629                        왓칭  20182501  2019-04-17   
628                     다시, 봄  20188393  2019-04-17   
627                       미성년  20180825  2019-04-11   
625                      한강에게  20183345  2019-04-04   
610                    질투의 역사  20183085  2019-03-14   
608                     국경의 왕  20176006  2019-02-28   
607                   칠곡 가시나들  20185350  2019-02-27   
603                항거:유관순 이야기  20196601  2019-02-27   
597      극장판 헬로카봇: 옴파로스 섬의 비밀  20185478  2019-01-31   
596                        이월  20173993  2019-01-30   
595                        뺑반  20180868  2019-01-30   
587       극장판 공룡메카드: 타이니소어의 섬  20183915  2019-01-10   
584                        언니  20184187  2019-01-01   
576            안녕, 나의 소녀 시절이여  20181876  2018-12-12   
575                    트와이스랜드  20184529  2018-12-07   
570                       도어락  20179230  2018-12-05   
571             런닝맨 : 풀룰루의 역습  20183745  2018-12-05   
566                   국가부도의 날  20177552  2018-11-28   
562                   뷰티풀 데이즈  20174442  2018-11-21   
560                      마담 B  20162979  2018-11-15   
559                      인어전설  20174008  2018-11-15   
556                       여곡성  20179006  2018-11-08   
545                       풀잎들  20180401  2018-10-25   
532                   죄 많은 소녀  20173734  2018-09-13   
524                      상류사회  20177537  2018-08-29   
522                    너의 결혼식  20153446  2018-08-22   
517                        22  20164048  2018-08-14   
511     신비아파트: 금빛 도깨비와 비밀의 동굴  20185242  2018-07-25   
509                       박화영  20171681  2018-07-19   
507                      속닥속닥  20180282  2018-07-13   
500                        마녀  20185341  2018-06-27   
497                      여중생A  20184101  2018-06-20   
494                       데자뷰  20173865  2018-05-30   
491                      오목소녀  20183423  2018-05-24   
486                 임을 위한 행진곡  20182323  2018-05-16   
485                        해원  20176025  2018-05-10   
482                  클레어의 카메라  20176101  2018-04-25   
478                    나를 기억해  20177266  2018-04-19   
476                    당신의 부탁  20177962  2018-04-19   
475                       스파키  20182221  2018-04-12   
473            호랑이보다 무서운 겨울손님  20174010  2018-04-12   
451                       게이트  20189663  2018-02-28   
448                       환절기  20162361  2018-02-22   
450                        라라  20180063  2018-02-22   
445       터닝메카드W: 반다인의 비밀 특별판  20189761  2018-02-08   
440                        천화  20166083  2018-01-25   
432                    B급 며느리  20174323  2018-01-17   
423                    련희와 연희  20170442  2017-12-14   
421           뽀로로 극장판 공룡섬 대모험  20174846  2017-12-07   
419                        초행  20174601  2017-12-07   
417                 아들에게 가는 길  20163906  2017-11-30   
416                       실종2  20174199  2017-11-30   
406                        채비  20175441  2017-11-09   
405                        미옥  20161681  2017-11-09   
404                  고향이 어디세요  20174983  2017-11-09   
390                    가을 우체국  20174641  2017-10-19   
381                    이웃집 스타  20172964  2017-09-21   
378                     사월의 끝  20163842  2017-09-14   
377            귀향, 끝나지 않은 이야기  20171164  2017-09-14   
366        

결정할 것

* 사용할 컬럼 종류
-> 영화명, 개봉년도, (장르: 따로 표시), (F-Rated: 따로 표시), 감독, 작가, 배우

* 여성과 남성이 함께 감독을 하거나 각본을 쓴 영화 포함? (79번 '오늘영화', 501번 허스토리)  
-> 포함하고 여성 이름에 하이라이트

* 단편영화 묶인 것들 어떻게 해야하나 (657번 한낮의 피크닉, 646번 우리 지금 만나 등)  
-> 단편은 정보가 다 있으면 각각 떼어서 넣기

* 다큐멘터리, 애니메이션, 공연실황 포함? 의미있는 작품들만 포함할까?  
-> 장르별로 인덱싱이 의미있을 것 같다 (다큐/액션/코미디/드라마/스릴러/공포)  
-> 아동용 애니메이션 제외

* 창작자가 불미스러운 사건에 휘말려 있는 경우의 영화 포함?  
-> 상관 없이

* 정렬: 최근 개봉작 (연도만 쓰기)